src/main/java/uk/ac/cam/cl/dtg/isaac/dao/PgAssignmentPersistenceManager.java

```diff
@@ -140,25 +140,7 @@ public AssignmentDTO getAssignmentById(final Long assignmentId) throws SegueData
 
     @Override
     public List<AssignmentDTO> getAssignmentsByGroupId(final Long groupId) throws SegueDatabaseException {
-        try (Connection conn = database.getDatabaseConnection()) {
-            PreparedStatement pst;
-            pst = conn.prepareStatement("SELECT * FROM assignments WHERE group_id = ? ORDER BY creation_date");
-
-            pst.setLong(1, groupId);
-            
-            ResultSet results = pst.executeQuery();
- 
-            List<AssignmentDTO> listOfResults = Lists.newArrayList();
-            
-            while (results.next()) {
-                listOfResults.add(this.convertToAssignmentDTO(this.convertFromSQLToAssignmentDO(results)));
-            }
-
-            return listOfResults;
-
-        } catch (SQLException e) {
-            throw new SegueDatabaseException("Unable to find assignment by group", e);
-        }
+        return this.getAssignmentsByGroupList(Collections.singletonList(groupId));
```

Comment:
I wondered if using the `group_id IN (...)` syntax may be less efficient in Postgres, but the query planner optimises it away.
On the other hand, it does perform a sequential scan. I have added an additional index to the live site to speed this up!